### PR TITLE
Add SQL functions CURRENT_DATE, CURRENT_TIME and CURRENT_TIMESTAMP

### PR DESCRIPTION
**Pull request by vaijns**
I guess you can also just use std::chrono instead but with those functions (which are also standardized) it might feel more like SQL.

-Jonas

### CODE REVIEW

**Comment by rbock** [comment]
Hi Jonas,

Thanks for adding these functions. Seems very reasonable to me.

Would be great if you could add serialization tests, e.g. similar to https://github.com/rbock/sqlpp11/blob/main/tests/core/serialize/Upper.cpp

Note: In the optional-no-dynamic branch (major overhaul), I decided to drop the `_auto_alias_t`. We might want to drop it for newly added functions before adding them. WDYT?

Cheers,
Roland

**Comment by vaijns** [comment]
Hi Roland,

I've added some basic serialization tests, let me know if there's more to test but I guess the functions themselves are very basic as well, so I couldn't think of anything else.

I've mostly looked at the trim()-function, that's where the `_auto_alias_t` came from but I agree that it doesn't seem necessary.

-Jonas

On another note: I've thought about adding more date and time functions, namely `year_of`, `month_of`, ... (or maybe `extract` instead of those), `datediff`, `date_add`, `date_sub`. Is that something you'd be willing to merge or do you think this should rather be done in application code? One problem with this is not being standardized functions of course.
If you're interested I'd create an issue for it to discuss design details.

File: include/sqlpp11/current_date.h [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+          return current_date;
+        }
+        const T& operator()() const
+        {
+          return current_date;
+        }
+      };
+    };
+
+    constexpr current_date_t()
```
Comment (rbock) [inline include/sqlpp11/current_date.h:60]: Can you make this `default`?

File: include/sqlpp11/current_date.h [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+namespace sqlpp
+{
+  struct current_date_t : public expression_operators<current_date_t, day_point>, public alias_operators<current_date_t>
+  {
+    using _traits = make_traits<day_point, tag::is_expression, tag::is_selectable>;
+
+    using _nodes = detail::type_vector<>;
+    using _can_be_null = std::false_type;
+
+    struct _alias_t
```
Comment (rbock) [inline include/sqlpp11/current_date.h:41]: Do you actually need this? I would assume that you only need this for `select(sqlpp::current_date)`?

In the next generation of the library, I would (most likely) say this has to `select(sqlpp::current_date.as(some_alias))`.

One of the reasons for this is that I am generally removing some implicit logic (including implicit names). Another reason is code complexity. Automatic names make things harder...

**Review by rbock — COMMENTED** [review]
Thanks for the update including the tests! Added two more comments, that apply to all functions.

**Comment by rbock** [comment]
Regarding the proposed additional functions: Please open a separate issue.

File: include/sqlpp11/current_date.h (hunk 2 above)
Comment (vaijns) [reply include/sqlpp11/current_date.h]: Requiring explicit aliases seems reasonable. I didn't really know what/where `_nodes` and `_can_be_null` are used for and didn't know they were optional. Removed them now.

**Comment by rbock** [comment]
Hi Jonas,

Thanks for the cleanup. This looks good to me.

Merging...

Cheers,
Roland

**Pull request merged**